### PR TITLE
Batch updates for issues

### PR DESCRIPTION
**Pull request by ethantkoenig**
#868

Updates the 
- `:reponame/issues/:index/label`
- `:reponame/issues/:index/milestone`
- `:reponame/issues/:index/assignee`

endpoints to allow for multiple issues.

Also adds a front-end for opening/closing/assigning/labelling/milestoning multiple issues at once.

Screenshot

I am by no means a front-end expert, so any feedback is welcome :smile:

### CODE REVIEW

**Comment by bkcsoft** [comment]
I don't like having `Add Label` and `Remove Label` split... Otherwise I have no comments _on the design_, L-G-T-M :)

File: routers/repo/issue.go [hunk 1]
```diff
@@ -716,54 +717,84 @@ func UpdateIssueContent(ctx *context.Context) {
...
 }
 
-// UpdateIssueMilestone change issue's milestone
-func UpdateIssueMilestone(ctx *context.Context) {
-	issue := getActionIssue(ctx)
-	if ctx.Written() {
-		return
-	}
+// UpdateIssues perform update on collection of issues
+func UpdateIssues(ctx *context.Context) {
```
Comment (bkcsoft) [inline routers/repo/issue.go]: Ooh god please no 😱

**Review by bkcsoft — CHANGES_REQUESTED** [review]
Please don't do this 😢

File: routers/repo/issue.go (hunk 1 above)
Comment (ethantkoenig) [reply routers/repo/issue.go]: So do you want
- separate functions for milestones/labels/assignees, each of which can handle both a single or multiple issues?
- keep the pre-existing functions milestones/labels/assignees (which can only handle a single issue), and introduce 3 new milestones/labels/assignees for multiple issues?

Also, what is wrong with what I did? I don't mean to argue, I just want to learn.

Comment (bkcsoft) [reply routers/repo/issue.go]: Preferably the first one. I'm not read up on the specific functions to give any advice, but I could have a look sometime during the week if you need pointers 🙂 

The part that was wrong (in my opinion), was taking 3 small and clean functions, and turning them into one huge, hard to read switch-statement 🙂

**Comment by ethantkoenig** [comment]
@bkcsoft I have addressed your concerns (combine `Add Labels` and `Remove Labels` menus, split `UpdateIssues` back into multiple functions)

File: public/js/index.js [hunk 2]
```diff
@@ -87,6 +87,20 @@ function initEditForm() {
...
 
+function updateIssuesMeta(url, action, issueIds, elementId, afterSuccess) {
+    $.ajax({
+        type: "POST",
+        url: url,
+        data: {
+        "_csrf": csrf,
+        "action": action,
+        "issue_ids": issueIds,
+        "id": elementId
```
Comment (andreynering) [inline public/js/index.js]: Missing indentation of `data` object

File: public/js/index.js [hunk 3]
```diff
@@ -119,13 +125,15 @@ function initCommentForm() {
             $(this).removeClass('checked');
             $(this).find('.octicon').removeClass('octicon-check');
             if (hasLabelUpdateAction) {
-                updateIssueMeta($labelMenu.data('update-url'), "detach", $(this).data('id'));
+                updateIssuesMeta($labelMenu.data('update-url'), "detach",
+                    $labelMenu.data('issue-id'), $(this).data('id'));
```
Comment (andreynering) [inline public/js/index.js]: I think, or keep function call on a single line, or keep each param in a line, like this:

```js
updateIssuesMeta(
  $labelMenu.data('update-url'),
  "detach",
  $labelMenu.data('issue-id'),
  $(this).data('id')
);
```

Same for similar cases below.

**Review by andreynering — APPROVED** [review]
Just two little sintax issues.

**Comment by andreynering** [comment]
Useful feature. LGTM

Ref: https://github.com/gogits/gogs/issues/2049

**Comment by ethantkoenig** [comment]
Rebasing to resolve conflicts.

**Comment by lunny** [comment]
conflicted.

**Comment by ethantkoenig** [comment]
Rebased.

File: routers/repo/issue.go [hunk 4]
```diff
@@ -644,6 +645,25 @@ func getActionIssue(ctx *context.Context) *models.Issue {
 	return issue
 }
 
+func getActionIssues(ctx *context.Context) []*models.Issue {
+	commaSeparatedIssueIDs := ctx.Query("issue_ids")
+	if len(commaSeparatedIssueIDs) == 0 {
+		return nil
+	}
+	stringIssueIDs := strings.Split(commaSeparatedIssueIDs, ",")
```
Comment (lunny) [inline routers/repo/issue.go]: Why not use `x.In().Find()`? It's better performance?

File: routers/repo/issue.go [hunk 5]
```diff
@@ -725,24 +742,52 @@ func UpdateIssueMilestone(ctx *context.Context) {
...
+	switch action := ctx.Query("action"); action {
+	case "open":
+		isClosed = false
+	case "close":
+		isClosed = true
+	default:
+		log.Warn("Unrecognized action: %s", action)
+	}
+
+	for _, issue := range issues {
```
Comment (lunny) [inline routers/repo/issue.go]:
```
IssueList(issues).LoadRepositories()
```

**Comment by lunny** [comment]
otherwise LGTM

File: routers/repo/issue.go (hunk 4 above)
Comment (ethantkoenig) [reply routers/repo/issue.go]: Fixed.

File: routers/repo/issue.go (hunk 5 above)
Comment (ethantkoenig) [reply routers/repo/issue.go]: Fixed.

**Comment by ethantkoenig** [comment]
@bkcsoft I've addressed your concerns from a few weeks ago; please re-review when you get the chance :smile:

**Comment by lunny** [comment]
but seems you changed index.css file and not index.less ?

**Comment by ethantkoenig** [comment]
Oh, didn't realize there was a `.less` file. Will update.

**Comment by ethantkoenig** [comment]
@lunny Updated to use `_repository.less`.

**Comment by lunny** [comment]
build failed and need @bkcsoft confirmation.

**Review by bkcsoft — APPROVED** [review]

**Comment by bkcsoft** [comment]
LGTM

**Pull request merged**